packages/docusaurus/bin/beforeCli.mjs

```diff
@@ -16,16 +16,24 @@ import boxen from 'boxen';
 import {createRequire} from 'module';
 
 const packageJson = createRequire(import.meta.url)('../package.json');
-const sitePkg = createRequire(path.join(process.cwd(), 'package.json'))(
-  './package.json',
-);
+/** @type {Record<string, any>} */
+let sitePkg;
+try {
+  sitePkg = createRequire(path.resolve('package.json'))('./package.json');
+} catch {
+  logger.warn`path=${'package.json'} file not found at CWD: path=${process.cwd()}.`;
+  logger.info`This is non-critical, but could lead to undesired behavior downstream. Docusaurus assumes that path=${'package.json'} exists at CWD, because it's where the package manager looks up the script at. A common reason is because you have changed directory in the script. Instead of writing code=${'"start": "cd website && docusaurus start"'}, consider using the code=${'[siteDir]'} argument: code=${'"start": "docusaurus start website"'}.`;
+  sitePkg = {};
```

Comment:
I don't really like this 😅 

either we support it properly, or fail-fast?

Reply:
1. I don't want to "support it properly", namely, using `pkg-up` to try to find _any_ package.json in the path, because that opens the doors to more edge-cases and makes future work harder.
2. Fail-fast _could_ work, but do we really want to fail-fast for something non-critical?

Reply:
ok

I don't really like to use `sitePkg = {}` as fallback, why not make things explicit then and use undefined?

Reply:
Because we are then destructuring `version` and `dependencies` etc. from it. Where do you want `undefined`?

Reply:
sitePkg = undefined

then handle the rest with if (sitePkg) { ... do things }

Reply:
I mean, what benefits does that bring?😅 We don't really differentiate the case between "no package" and "a package without dependencies". Both lead to an empty field anyways.

Reply:
Yes, if we do have such case in the future we will change that; right now the semantic is not any different so I'd rather not complicate it.